snuba/utils/streams/configuration_builder.py

```diff
@@ -32,14 +41,15 @@ def get_default_kafka_configuration(
 def build_kafka_consumer_configuration(
     topic: Optional[Topic],
     group_id: str,
+    slice_id: Optional[int] = None,
     auto_offset_reset: Optional[str] = None,
     queued_max_messages_kbytes: Optional[int] = None,
     queued_min_messages: Optional[int] = None,
     bootstrap_servers: Optional[Sequence[str]] = None,
     override_params: Optional[Mapping[str, Any]] = None,
     strict_offset_reset: Optional[bool] = None,
 ) -> KafkaBrokerConfig:
-    default_topic_config = _get_default_topic_configuration(topic)
+    default_topic_config = _get_default_topic_configuration(topic, slice_id)
```

Comment:
You can simplify this a bit. This logic can be moved out to the caller and you only pass in the topic config to this api. So you don't need to pass in `topic` and `slice_id` here. This just reduces the amount of parameters which are being passed around.

Reply:
If I understand what you are suggesting, I can move the logic inside `_get_default_topic_configuration` into `get_default_kafka_configuration` itself. Only thing is this would remove our ability to separately test the mapping functionality in `_get_default_topic_configuration`.